File: guides/spring-boot-to-micronaut-testing-rest-api-integrations-using-mockserver/micronautframework/java/src/main/java/example/micronaut/AlbumController.java
package example.micronaut;

import io.micronaut.http.HttpResponse;
import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Get;
import io.micronaut.http.annotation.PathVariable;
import java.util.List;

@Controller("/api")
class AlbumController {
    private final PhotoServiceClient photoServiceClient;

    AlbumController(PhotoServiceClient photoServiceClient) {
        this.photoServiceClient = photoServiceClient;
    }

    @Get("/albums/{albumId}")
    public HttpResponse<Album> getAlbumById(@PathVariable Long albumId) {
        List<Photo> photos = photoServiceClient.getPhotos(albumId);
        return HttpResponse.ok(new Album(albumId, photos));
    }
}

File: guides/testing-rest-api-integrations-using-mockserver/python/src/example/micronaut/album_controller.py
from typing import Annotated

from micronaut.http.annotation import Controller, Get, PathVariable
from micronaut.scheduling import TaskExecutors
from micronaut.scheduling.annotation import ExecuteOn

from .album import Album
from .photo_service_client import PhotoServiceClient


@Controller("/api")  # <1>
class AlbumController:
    def __init__(self, photo_service_client: PhotoServiceClient):  # <2>
        self.photo_service_client = photo_service_client

    @ExecuteOn(TaskExecutors.BLOCKING)  # <3>
    @Get("/albums/{albumId}")  # <4>
    def get_album_by_id(self, albumId: Annotated[int, PathVariable]) -> Album:  # <5>
        return Album(albumId, self.photo_service_client.get_photos(albumId) or [])
